From: Juergen Gross <jgross@suse.com>
Date: Tue, 25 Mar 2025 13:39:15 +0200
Subject: kABI: Fix kABI after KVM: x86: Snapshot the host's DEBUGCTL in common x86
Patch-mainline: Never, kABI fix
References: jsc#PED-6143

Move host_debugctl in struct kvm_vcpu_arch into a mostly unused hole
and the sole bool in that hole into another one.

Signed-off-by: Juergen Gross <jgross@suse.com>
---
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -760,10 +760,11 @@
 	u32 pkru;
 	u32 hflags;
 	u64 efer;
-	u64 host_debugctl;
 	u64 apic_base;
 	struct kvm_lapic *apic;    /* kernel irqchip context */
+#ifndef __GENKSYMS__
-	bool load_eoi_exitmap_pending;
+	u64 host_debugctl;
+#endif
 	DECLARE_BITMAP(ioapic_handled_vectors, 256);
 	unsigned long apic_attention;
 	int32_t apic_arb_prio;
@@ -774,6 +775,9 @@
 	bool at_instruction_boundary;
 	bool tpr_access_reporting;
 	bool xfd_no_write_intercept;
+#ifndef __GENKSYMS__
+	bool load_eoi_exitmap_pending;
+#endif
 	u64 ia32_xss;
 	u64 microcode_version;
 	u64 arch_capabilities;
